apps/mitt-konto/src/Components/Paywall.purs

```diff
@@ -0,0 +1,217 @@
+        , DOM.button
+            { children: [ DOM.text "Skicka" ]
+            , onClick: submitHandler
+            , disabled: days + hours + minutes == 0 || null selection
+            }
+        , DOM.hr {}
+        , DOM.h2_ [ DOM.text "Nuvarande betalmur öppningar" ]
+        , maybe loadingSpinner (renderCurrentOpenings setEra) openings
+        ]
+      }
+
+setNumber :: Int -> (Int -> Effect Unit) -> EventHandler
+setNumber max setter = handler targetValue \value ->
+  maybe (pure unit) setter (value >>= handleEmpty >>= fromString >>= handleBounds)
+  where
+    handleEmpty s = Just case s of
+      "" -> "0"
+      _  -> s
+    handleBounds n
+      | n < 1     = Just 0
+      | n > max   = Just max
+      | otherwise = Just n
+
+renderDays :: Int -> (Int -> Effect Unit) -> JSX
+renderDays days setDays =
+  DOM.label
+    { children:
+        [ DOM.input
+            { type: "text"
+            , value: show days
+            , onChange: setNumber 365 setDays
+            , style: DOM.css { width: "2rem" }
+            }
+        , DOM.text " dagar"
+        ]
+      , style: DOM.css { marginRight: "1rem" }
+    }
+
+renderHours :: Int -> (Int -> Effect Unit) -> JSX
+renderHours hours setHours =
+  DOM.label
+    { children:
+        [ DOM.input
+            { type: "text"
+            , value: show hours
+            , onChange: setNumber 23 setHours
+            , style: DOM.css { width: "2rem" }
+            }
+        , DOM.text " timmar"
+        ]
+      , style: DOM.css { marginRight: "1rem" }
+    }
+
+renderMinutes :: Int -> (Int -> Effect Unit) -> JSX
+renderMinutes minutes setMinutes =
+  DOM.label
+    { children:
+        [ DOM.input
+            { type: "text"
+            , value: show minutes
+            , onChange: setNumber 59 setMinutes
+            , style: DOM.css { width: "2rem" }
+            }
+        , DOM.text " minuter"
+        ]
+      , style: DOM.css { marginRight: "1rem" }
+    }
+
+toggleProduct :: String -> Products -> Products
+toggleProduct name products =
+  fromMaybe initialProducts $ do
+    index <- findIndex (\p -> p.name == name) products
+    let modify p = p { selected = not p.selected }
+    modifyAt index modify products
+
+renderProducts
+  :: Products
+  -> ((Products -> Products) -> Effect Unit)
+  -> JSX
+renderProducts products setProducts =
+  DOM.dl
+    { children:
+        DOM.dt_ [ DOM.text "produkter"] : map renderProduct products
+    , style: DOM.css { listStyleType: "none" }
+    }
+  where
+    renderProduct { name, label, selected } =
+      DOM.dl_
+      [ DOM.label
+          { children:
+              [ DOM.input
+                  { type: "checkbox"
+                  , value: name
+                  , checked: selected
+                  , onChange: mkEffectFn1 (\_ -> setProducts (toggleProduct name))
+                  }
+              , DOM.text label
+              ]
+          }
+      ]
+
+renderCurrentOpenings
+  :: ((Int -> Int) -> Effect Unit)
```

Comment:
I would suggest changing this to `Int -> EventHandler` where `Int` is the id of the line and have the function definition on lines 189-191 be inside the let block starting on line 51.